### ISSUE TITLE
Behavior 'Expand All' with angular animate 1.3.15 and ui-grid 3.0.0-RC21 

### ISSUE THREAD

**Issue by Jacquelin** [angular-ui/ng-grid#3696]
Hi,

I'm not sure why, and how, but the expand all action work fine, but the icons are not good.
http://plnkr.co/edit/URYETKsjgPUOgFKPhfYK?p=preview

If I disable ngAnimate, it's working fine.
Is a duplicate of #3647 ?

**Comment by PaulL1** [comment]
I think it's associated with using groupPriority: 100 - in general groupPriority is numeric starting at 1:
http://plnkr.co/edit/c7PomebeNm6pUuo8uunv?p=preview

Having said that, I thought we'd done tidyup on the priorities on initialisation to avoid issues like this.

**Comment by Jacquelin** [comment]
Hi Paul

it seems to be related to ngAnimate. if I remove ngAnimate in module app, there is no problem.

**Comment by PaulL1** [comment]
Yeah, the problem is that that makes no sense.  There's no animation code anywhere in grouping.  Which leaves me wondering what else is broken.  If you downgrade to angular 1.2 then it works fine with animation still included.

**Comment by Jacquelin** [comment]
Hi,

Yes, if I downdrage to 1.2 it's working, but we can't downgrade, we have already develop every thing in 1.3 ...

I read a bit about ngAnimate, https://docs.angularjs.org/api/ngAnimate, and even if expand/collapseAll doesn't involve animate, ngClass does ! :

```
<i ng-class="{'ui-grid-icon-minus-squared': ( (  [...]
```

The conditions for ngClass are good! but I have the impression that the ngAnimate struggling with the transition from one class to another

For exemple, this condition return false
```
( ( grid.options.showTreeExpandNoChildren && row.treeLevel > -1 ) || ( row.treeNode.children && row.treeNode.children.length > 0 ) ) && row.treeNode.state === 'expanded'
```
But the <i> have class="ui-grid-icon-minus-squared" ...

I continue to investigate, hoping to find a solution

**Comment by Jacquelin** [comment]
@PaulL1 : I have an idea but I do not know if it's worth it. You can disable ngAnimate only for uiGridTreeBaseRowHeaderButtons Directive. This involves adding

```
$ animate.enabled (false, $ elm); // line ~1490 on tree-base.js
```

what do you think about it?

**Comment by PaulL1** [comment]
Rumour has it that 1.3.16 works.  Is that worth trying?

**Comment by Jacquelin** [comment]
Hello,

I try with angular 1.3.16 but ... It doesn't work :( (http://plnkr.co/edit/URYETKsjgPUOgFKPhfYK?p=preview)

I try with angular 1.4.0 and ... it work lol

**Comment by JLLeitschuh** [comment]
You may have other problems with angular 1.4.0 #3658

**Comment by postrational** [comment]
I am seeing a similar problem, as illustrated in the image below:

![grouping-bug](https://cloud.githubusercontent.com/assets/4430709/8305317/18a1518c-19ae-11e5-8e94-4aba028497d4.png)

Rows which represent entries in a group sometimes also display the group's minus symbol.

I created a Plunker. You may need to collapse and expand the rows a few times to see the bug:
http://embed.plnkr.co/XraGUgxK0sQfP6plsNnP/preview

**Comment by klawson23** [comment]
Hey everyone, I just wanted to bump this issue and say that I am also experiencing this problem exactly as described with angular 1.3.15 and ui-grid 3.0.0-RC22. 

It doesn't appear to break anything functionally, and clicking the extra minus symbols doesn't appear to do anything but it doesn't look great. 

Is updating to the latest version of angular the recommended solution or have people encountered other issues with angular 1.4.2 and ui-grid as @JLLeitschuh mentioned?

**Comment by JLLeitschuh** [comment]
The animate problem is due to a flip in the parameters for animate:
http://brianhann.com/angular-1-4-breaking-changes-to-be-aware-of/
We are in the process of updating ui-grid to support 1.4.
